shopinvader_delivery_carrier/models/delivery_carrier.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from contextlib import contextmanager
+
+from odoo import api, models
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = "delivery.carrier"
+
+    @contextmanager
+    def _simulate_delivery_cost(self, partner):
+        """
+        Change the env mode (draft) to avoid real update on the partner.
+        Then, restore the partner with previous values.
+        :param partner: res.partner recordset
+        :return:
+        """
+        partner.read()
+        partner_values = partner._convert_to_write(partner._cache)
+        with partner.env.do_in_draft():
```

Comment:
@acsonefho We saw this writes to DB. So, ...

Reply:
It seems that the issue doesn't appears in this case.
Unit test go into this code part and if you execute a sql query, the country/zip still correct, even if it's updated into the cache.